tools/python/create_nifi_reporting_task.py

```diff
@@ -0,0 +1,155 @@
+        revision=nipyapi.nifi.models.revision_dto.RevisionDTO(version=0),
+        disconnected_node_acknowledged=False,
+        component=nipyapi.nifi.models.reporting_task_dto.ReportingTaskDTO(
+            name=name,
+            type="org.apache.nifi.reporting.prometheus.PrometheusReportingTask",
+            bundle=nipyapi.nifi.models.bundle_dto.BundleDTO(
+                group="org.apache.nifi",
+                artifact="nifi-prometheus-nar",
+                version=version
+            ),
+            properties={
+                "prometheus-reporting-task-metrics-endpoint-port": port,
+                "prometheus-reporting-task-metrics-send-jvm": True
+            }
+        )
+    )
+
+    controller_api = nipyapi.nifi.apis.controller_api.ControllerApi()
+
+    try:
+        return controller_api.create_reporting_task(body=task)
+    except Exception as ex:
+        raise Exception("Could not create reporting task: {}".format(str(ex))) from None
+
+
+def get_reporting_task_id(task):
+    """Return the ReportingTask ID"""
+    return task.id
+
+
+def get_reporting_task_name(task):
+    """Return the ReportingTask name"""
+    task_dict = task.to_dict()
+    return task_dict["component"]["name"]
+
+
+def get_revision_version(task):
+    """Return the ReportingTask revision version"""
+    task_dict = task.to_dict()
+    return task_dict["revision"]["version"]
+
+
+def is_reporting_task_running(task):
+    """Check if the the ReportingTask is already running"""
+    task_dict = task.to_dict()
+    return task_dict["component"]["state"] == "RUNNING"
+
+
+def set_reporting_task_running(task):
+    """Set ReportingTask to RUNNING"""
+    reporting_task_api = nipyapi.nifi.apis.reporting_tasks_api.ReportingTasksApi()
+
+    state = {
+        "revision": {
+            "version": get_revision_version(task)
+        },
+        "disconnected_node_acknowledged": False,
+        "state": "RUNNING"
+    }
+
+    try:
+        return reporting_task_api.update_run_status(id=get_reporting_task_id(task), body=state)
+    except Exception as ex:
+        raise Exception("Could not set ReportingTask [{}] to RUNNING: {}"
+                        .format(get_reporting_task_id(task), str(ex))) from None
+
+
+def main():
+    """Main method with cli argument parsing and ReportingTask logic"""
+    # Construct an argument parser
+    all_args = argparse.ArgumentParser()
+    # Add arguments to the parser
+    all_args.add_argument("-n", "--nifi_api_url", required=True,
+                          help="The NiFi node url to connect to.")
+    all_args.add_argument("-u", "--username", required=True,
+                          help="Username to connect as.")
+    all_args.add_argument("-p", "--password", required=True,
+                          help="Password for the user.")
+    all_args.add_argument("-v", "--nifi_version", required=True,
+                          help="The NiFi product version.")
+    all_args.add_argument("-c", "--cert", required=True,
```

Comment:
This could be made a bit clearer .. it is not any public certificate, it is  the certificate authority that we trust and which has to have signed the certificates that we expect..

Reply:
done